.circleci/config.yml

```diff
@@ -149,3 +168,10 @@ workflows:
             branches:
               ignore:
                 - main
+      - build_deploy_simple_api:
```

Comment:
Ah I just realized that I actually only want to run this when the PR is merged. Perhaps either of you have an idea of how to configure the script for that? I'll do some more digging.

Reply:
@adschwartz what behaviour do you want exactly? I'm guessing maybe you want this job to run only on main (since you're publishing to docker), but at the same time you mention `branched > ignore > main`, so not sure.
If what you want is really a one-off thing to start running it when this PR merges, what I'd do would be to set a branch filter to run it ONLY on main for this PR, and then when this PR is merged, you can remove the filter to run it everywhere (or start ignoring main if you want to run it only on PR branches). At least that's how I'd do it

Reply:
That being said, I think a pretty common pattern is to split the "build" and "publish" phase of a job. Here your `build_deploy_simple_api` is doing both. I think if you split it, at least you'd be able to run the 'build' phase on PRs everywhere, and then you can activate the publish part either later, or only on main (i.e. which means it will start running when this PR is merged, but only on main).

Reply:
💯  to what Guillaume said; could use https://github.com/kurtosis-tech/kurtosis/blob/main/.circleci/config.yml for inspiration (though in this case publish happens on tag; you can change it to publish happening on merge to main)

build - the ignore main runs on PRs and random branches
publish - while the one set to run on main only run when the merge happens (as thats the only way to write to main)

Reply:
@gbouv My intention was to only build and publish on merge to main. But I can see the value in building if there's a PR and then build and publish on merge to main. I was looking for examples of us doing this but didn't find any. As @h4ck3rk3y mentioned we run jobs based on the presence of tags I believe. I was googling this a bit yesterday and didn't find a simple way to filter on if there's a merge or not. But I could be overlooking it and was looking to see if you two knew how to do it.

Reply:
Any push to main is a merge as our repository is protected. I'd replace the tags in the publish jobs to main like 

```
      - publish_docker_image
          context:
            - docker-user
          filters:
            branches:
              only: "main"
```

Reply:
@adschwartz let me know if I'm missing something but I think you can achieve this with something like:
```
jobs:
  build_simple_api:
    executor: ubuntu_vm
    steps:
      - checkout
      # If this sub-repo hasn't changed, don't run the tests
      - run: |
          if git --no-pager diff --exit-code origin/main...HEAD -- ':.circleci' ':simple-api'; then
            circleci-agent step halt
          fi
      - run: simple-api/scripts/build.sh

  publish_simple_api:
    executor: ubuntu_vm
    steps:
      - checkout
      # If this sub-repo hasn't changed, don't run the tests
      - run: |
          if git --no-pager diff --exit-code origin/main...HEAD -- ':.circleci' ':simple-api'; then
            circleci-agent step halt
          fi
      - run: simple-api/scripts/build.sh
      - run: echo "${DOCKER_PASSWORD}" | docker login -u ${DOCKER_USERNAME} --password-stdin
      - run: |
          set -euo pipefail
          version_to_publish="latest"
          image_org_and_repo="kurtosistech/kurtosis-simple-api"
          image_name_with_version="${image_org_and_repo}:${version_to_publish}"
          docker push "${image_name_with_version}"
workflows:
  build:
    jobs:
      - build_simple_api:
          filters:
            branches:
              ignore:
                - main
      - publish_simple_api:
          filters:
            branches:
              only:
                - main
```

Reply:
@h4ck3rk3y @gbouv wow you guys rock. Didn't expect you to hand me the solution, but really appreciate it!

Reply:
Thank you both for the inputs here. Appreciated. @gbouv @h4ck3rk3y